security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java

```diff
@@ -37,14 +42,26 @@ public AnonymousInterceptorWrapper() {
         //make sure this interceptor runs before the WSS4J one in the same Phase, otherwise it won't work
         Set<String> before = getBefore();
         before.add(WSS4JInInterceptor.class.getName());
+        before.add(PolicyBasedWSS4JInInterceptor.class.getName());
 	}
 
-    public void setAnonIntercep(PhaseInterceptor anonIntercep) {
-        this.anonIntercep = anonIntercep;
+    protected BundleContext getContext() {
+        Bundle cxfBundle = FrameworkUtil.getBundle(AnonymousInterceptorWrapper.class);
+        if (cxfBundle != null) {
+            return cxfBundle.getBundleContext();
+        }
+        return null;
     }
 
     @Override
 	public void handleMessage(SoapMessage msg) throws Fault {
+        BundleContext context = getContext();
+        PhaseInterceptor anonIntercep = null;
+        ServiceReference anonIntercepRef = context.getServiceReference(PhaseInterceptor.class.getName());
```

Comment:
Is there a reason we cannot do a service reference list for this?  Then we do not need the bundle context.